### PR TITLE
Reworked material management

### PR DESCRIPTION
**Pull request by favreau**

### CODE REVIEW

**Comment by tribal-tec** [comment]
your change shouldn't affect the visual appearance, so why the testdata has to be updated? Or it was broken before?

**Comment by tribal-tec** [comment]
The update of CMake/common is necessary? It's the latest version?

File: brayns/common/engine/Engine.h [hunk 1]
```diff
@@ -98,15 +98,16 @@ class Engine
 
     /**
        Initializes materials for the current scene
-       @param materialType Predefined sets of colors
+       @param colorMap Predefined color map
              MT_DEFAULT: Random colors
```
Comment (tribal-tec) [inline brayns/common/engine/Engine.h]: those MT_* values are no longer correct, no?

File: brayns/common/scene/Scene.cpp [hunk 2]
```diff
@@ -247,78 +235,78 @@ void Scene::buildDefault()
...
-        m.setColor(colors[material]);
-        m.setSpecularColor(WHITE);
-        m.setSpecularExponent(10.f);
-        m.setReflectionIndex(material == 4 ? 0.8f : 0.f);
-        m.setOpacity(1.f);
-        auto& trianglesMesh = _trianglesMeshes[NB_SYSTEM_MATERIALS + material];
-        for (size_t i = 0; i < 6; ++i)
+        // Cornell box
+        Material material;
+        material.setColor(colors[i]);
```
Comment (tribal-tec) [inline brayns/common/scene/Scene.cpp]: constructor for Material to take some/all of this values?

File: brayns/common/scene/Scene.cpp [hunk 3]
```diff
@@ -699,34 +699,32 @@ void Scene::saveToCacheFile()
...
         file.write((char*)&value, sizeof(float));
-        const bool boolean = material.second.getCastSimulationData();
+        const bool boolean = material.getCastSimulationData();
         file.write((char*)&boolean, sizeof(bool));
         // TODO: Textures
     }
 
     // Save geometry
-    for (auto& material : _materials)
+    ;
```
Comment (tribal-tec) [inline brayns/common/scene/Scene.cpp]: rm

File: brayns/common/scene/Scene.h [hunk 4]
```diff
@@ -360,7 +327,56 @@ class Scene
...
+    BRAYNS_API void setMaterial(const size_t index, const Material& material);
+
+    /**
+        Resets materials and builds the system ones (Bounding box, skymap, etc)
+    */
+    BRAYNS_API void resetMaterials();
+
+    /**
+        Commit materials to renderers
+        @param updateOnly If true, materials are not recreated and textures are
```
Comment (tribal-tec) [inline brayns/common/scene/Scene.h]: wrong doc

File: brayns/io/MeshLoader.cpp [hunk 5]
```diff
@@ -33,6 +33,9 @@
 
 #include <brayns/common/scene/Scene.h>
 
+#define BRAYNS_USE_ASSIMP \
+    1 //************* REMOVE BEFORE PUSH ******************
```
Comment (tribal-tec) [inline brayns/io/MeshLoader.cpp]: ??

File: brayns/common/engine/Engine.h (hunk 1 above)
Comment (favreau) [reply brayns/common/engine/Engine.h]: Doc is now in the types.h file. Removed.

File: brayns/common/scene/Scene.cpp (hunk 3 above)
Comment (favreau) [reply brayns/common/scene/Scene.cpp]: Done

File: brayns/common/scene/Scene.h (hunk 4 above)
Comment (favreau) [reply brayns/common/scene/Scene.h]: Done

File: brayns/io/MeshLoader.cpp (hunk 5 above)
Comment (favreau) [reply brayns/io/MeshLoader.cpp]: Comment was obliviously not big enough... :see_no_evil:

File: brayns/common/scene/Scene.cpp (hunk 2 above)
Comment (favreau) [reply brayns/common/scene/Scene.cpp]: hum... it's either all or none, some would be a bit weird to me. And having them all makes it a long list of arguments. Maybe another idea would to be to define some presets, like Metal, Glass, etc?

**Comment by favreau** [comment]
Regarding the new test file, the current one is broken and does now process material. This PR fixes the problem.

**Pull request merged**